### DISCUSSION TITLE
Subtitles

### DISCUSSION THREAD

**Discussion by l0rdbyte** [gerbera/gerbera#2822] [Q&A]
How?  I followed all the steps described in:   https://github.com/gerbera/gerbera/discussions/1102  None of those work.
The subs have the "name of videofile".srt format.  

On other DLNA implementations it worked fine.  Then you've got a great tip in the documentation about advanced configuration by enabling "expert' mode"... But nowhere is it mentioned HOW TO ACTUALLY ENABLE IT.

**Comment by l0rdbyte** [comment]
To clarify.  I've used multiple clients (vlc, dlna browser,...) on both home PCs and my Shield.  Integrated subtitles are no problem.

**Comment by l0rdbyte** [comment]
Also the "gerbera --create-config" function doesn't work... It doesn't exist.

**Comment by l0rdbyte** [comment]
[config.zip](https://github.com/gerbera/gerbera/files/11436803/config.zip)
Adding my config file.

**Comment by KarlStraussberger** [comment]
Seems you are using a really old version of gerbera. Check out `gerbera --version` and `gerbera --help`.

**Comment by l0rdbyte** [comment]
Gerbera UPnP Server version 1.1.0

Apparently only the last entry actually installs the latest version: 
`sudo add-apt-repository ppa:stephenczetty/gerbera-updates
sudo apt-get update
sudo apt install gerbera`

Now it's
Gerbera UPnP Server v1.12.1-37-g33c6554f
I removed the config, created one using gerbera --create-config
And now it won't even work:
The daemon says it's working. I can reach the first page but trying anything (with or without user login) gives a : ERR_CONNECTION_REFUSED

I've rebooted. Checked the config, checked the /usr/share/gerbera/web/  Changed users, changed home folders, changed web UI on or off, restarted the service a zillion times... 
Then just gave it rights because fuck it... Everything and the kitchen sink, and it still WILL NOT WORK.

● gerbera.service - Gerbera Media Server with MySQL
     Loaded: loaded (/lib/systemd/system/gerbera.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2023-05-19 00:03:51 CEST; 11min ago
   Main PID: 2602 (gerbera)
      Tasks: 2 (limit: 14140)
     Memory: 29.6M
        CPU: 70ms
     CGroup: /system.slice/gerbera.service
             └─2602 /usr/bin/gerbera -c /etc/gerbera/config.xml

May 19 00:03:51 skypirates systemd[1]: Started Gerbera Media Server with MySQL.
May 19 00:03:51 skypirates gerbera[2602]: 2023-05-19 00:03:51   info: Gerbera UPnP Server v1.12.1-37-g33c6554f - http://gerbera.io/
May 19 00:03:51 skypirates gerbera[2602]: 2023-05-19 00:03:51   info: Copyright 2016-2023 Gerbera Contributors.
May 19 00:03:51 skypirates gerbera[2602]: 2023-05-19 00:03:51   info: Gerbera is free software, covered by the GNU General Public License version 2
May 19 00:03:51 skypirates gerbera[2602]: 2023-05-19 00:03:51   info: Loading configuration from: /etc/gerbera/config.xml
May 19 00:03:51 skypirates gerbera[2602]: 2023-05-19 00:03:51   info: Parsing configuration...
May 19 00:03:51 skypirates gerbera[2602]: 2023-05-19 00:03:51   info: Configuration load succeeded.
May 19 00:03:51 skypirates gerbera[2602]: 2023-05-19 00:03:51   info: Validating configuration...
May 19 00:03:51 skypirates gerbera[2602]: 2023-05-19 00:03:51   info: Configuration check succeeded.

I am starting to think that this, and by extension the linux community is just an elaborate hoax.  Check the help, check the tutorials, check whatever the fuck... and you get 17 contradictory statements, none of which work, in a ridiculously convoluted package, with zero user friendlyness or even useability features and I freaking come from the DOS age...  
I LH'ed and EMM and XMS jockeyed to get those few KB to run something.

**Reply by KarlStraussberger** [reply]
I gladly see that you are progressing in your efforts. When running into problems, I'd like to remind you that people working on this project don't get paid for and do all development and support for the thankful appreciation of the users.

For you startup problem there can be multiple issues, e.g. permissions to access files, network configuration, and so on. The easiest way to check on this is to run `gerbera --debug -c /etc/gerbera/config.xml` you should see a message like ` The Web UI can be reached by following this link: http:...` otherwise you may find some hints why it ceases to work. Be aware the the user you run this command with must have permissions to write to the `home` directory from your config file!

**Comment by l0rdbyte** [comment]
I am aware....  And for that thank you, especially your prompt help.  
Further progress:
If I run the debug it gives no faults, and to top it off, I can access the webpage, setup directories, and view the gerbera dlna server from my devices (and even stream).   And  restarting the service just gets it stuck on:
May 19 00:03:51 skypirates gerbera[2602]: 2023-05-19 00:03:51   info: Configuration check succeeded.
May 19 11:43:23 skypirates systemd[1]: Stopping Gerbera Media Server with MySQL...

It never showed the IP / webaddress as it normally does, so it seems it's stuck somewhere.  But it should have root access if it runs as a service.

**Reply by KarlStraussberger** [reply]
In that case I recommend checking the user which the service wants to use and the database file (I not sure whether migration from 1.1 really works). Also you should add your media directory directly in config.xml (section autoscan). With 1.12.1 you can also activate some debug-stuff in your config.xml `<server debug-mode="content|server|cds|autoscan|web">` or even add `--debug` in the startup of the service script.
